dbt/include/databricks/macros/adapters.sql

```diff
@@ -105,16 +115,234 @@
 {% endmacro %}
 
 {% macro databricks__persist_constraints(relation, model) %}
-  {# Model contracts are not currently supported. #}
+
+  {% set delta_format = config.get('file_format', 'delta') == 'delta' %}
+
   {%- set contract_config = config.get('contract') -%}
-  {% if contract_config and contract_config.enforced %}
-    {{ exceptions.raise_compiler_error('Model contracts are not currently supported.') }}
+  {% set has_model_contract = contract_config and contract_config.enforced %}
+
+  {% set has_databricks_constraints = config.get('persist_constraints', False) %}
+
+  {% if (has_model_contract or has_databricks_constraints) and delta_format and not is_incremental() %}
```

Comment:
is_incremental() here is weird. What happens if users change constraints but do not run full_refresh?

Reply:
Our previous implementation and the dbt implementation both skip adding table constraints if is_incremental() is true.
I moved the check to here rather than having it inside the loops in the individual functions.
Model contracts are supported on incremental materializations with on_schema_change=append_new_columns i.e. any changes would be additive.
So if the model sql changes to add new columns the model definition would have to be updated to include the new file and any constraints on the new file.

Reply:
We should log a warning for any constraint defined in models that are not DELTA.
We should log a warning that we did not update any constraints after incremental runs. If you want to update constraints, run full_refresh